internal/operators/manager.go

```diff
@@ -1,155 +1,281 @@
 package operators
 
 import (
+	"container/list"
+	"context"
 	"encoding/json"
+	"fmt"
 
 	"github.com/go-openapi/swag"
-	"github.com/kelseyhightower/envconfig"
 	"github.com/openshift/assisted-service/internal/common"
-	"github.com/openshift/assisted-service/internal/operators/lso"
-	"github.com/openshift/assisted-service/internal/operators/ocs"
+	"github.com/openshift/assisted-service/internal/operators/api"
 	"github.com/openshift/assisted-service/models"
 	"github.com/sirupsen/logrus"
 )
 
 // Manager is responsible for performing operations against additional operators
 type Manager struct {
-	log                logrus.FieldLogger
-	ocsValidatorConfig *ocs.Config
-	ocsValidator       ocs.OcsValidator
+	log       logrus.FieldLogger
+	operators map[models.OperatorType]api.Operator
 }
 
 //go:generate  mockgen -package=operators -destination=mock_operators_api.go . API
 type API interface {
-	// ValidateOCSRequirements validates OCS requirements
-	ValidateOCSRequirements(cluster *common.Cluster) string
+	// ValidateCluster validates cluster requirements
+	ValidateCluster(ctx context.Context, cluster *common.Cluster) ([]api.ValidationResult, error)
+	// ValidateHost validates host requirements
+	ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) ([]api.ValidationResult, error)
 	// GenerateManifests generates manifests for all enabled operators.
 	// Returns map assigning manifest content to its desired file name
 	GenerateManifests(cluster *common.Cluster) (map[string]string, error)
 	// AnyOperatorEnabled checks whether any operator has been enabled for the given cluster
 	AnyOperatorEnabled(cluster *common.Cluster) bool
-	// GetOperatorStatus gets status of an operator of given type.
-	GetOperatorStatus(cluster *common.Cluster, operatorType models.OperatorType) string
+	// UpdateDependencies amends the list of requested additional operators with any missing dependencies
+	UpdateDependencies(cluster *common.Cluster) error
 }
 
-// NewManager creates new instance of an Operator Manager
-func NewManager(log logrus.FieldLogger) Manager {
-	cfg := ocs.Config{}
-	err := envconfig.Process("myapp", &cfg)
-	if err != nil {
-		log.Fatal(err.Error())
-	}
-	return NewManagerWithConfig(log, &cfg)
-}
-
-// NewManagerWithConfig creates new instance of an Operator Manager
-func NewManagerWithConfig(log logrus.FieldLogger, cfg *ocs.Config) Manager {
-	ocsValidator := ocs.NewOCSValidator(log.WithField("pkg", "ocs-operator-state"), cfg)
-	return Manager{
-		log:                log,
-		ocsValidatorConfig: cfg,
-		ocsValidator:       ocsValidator,
+// AddOperators adds operator plugins to the manager
```

Comment:
I think you should remove this method - didn't see that you are using it

Reply:
Done.